ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -89,8 +89,9 @@ var _ = Context("Inside the default namespace", func() {
 		headPod := corev1.Pod{}
 		headPods := corev1.PodList{}
 		workerPods := corev1.PodList{}
-		workerFilterLabels := client.MatchingLabels{utils.RayClusterLabelKey: rayCluster.Name, utils.RayNodeGroupLabelKey: rayCluster.Spec.WorkerGroupSpecs[0].GroupName}
-		headFilterLabels := client.MatchingLabels{utils.RayClusterLabelKey: rayCluster.Name, utils.RayNodeGroupLabelKey: utils.RayNodeHeadGroupLabelValue}
+		workerFilters := common.RayClusterGroupPodsAssociationOptions(rayCluster, rayCluster.Spec.WorkerGroupSpecs[0].GroupName).ToListOptions()
+		headGroupFilters := common.RayClusterGroupPodsAssociationOptions(rayCluster, utils.RayNodeHeadGroupLabelValue).ToListOptions()
```

Comment:
Why do we need to have both `headGroupFilters` and `headFilters`? The primary goal of this refactoring is to maintain consistency in the association method by always using the same function to associate two Kubernetes resources. If there is no special reason, we should remove `headGroupFilters`.

Reply:
Hi @kevin85421, actually the original `headFilterLabels` were equivalent to `headGroupFilters`. The `headFilters` are newcomers.

I found that the `ray.io/group` is still needed for a head Pod due to this: https://github.com/ray-project/kuberay/blob/a87f9a6a6d8ead0ffd81dc823f2fd7f38f2af027/ray-operator/controllers/ray/common/pod.go#L585-L593

Shouldn't we still test `ray.io/group` for a head Pod? We can leave a warning that we should normally use `headFilters` instead of `headGroupFilters` to do associations. The `headGroupFilters` are only meant to test the `ray.io/group=headgroup`.

Reply:
> Shouldn't we still test ray.io/group for a head Pod?

We have already had unit tests for `ray.io/group`. It's better to remove `headGroupFilters` from the env tests.